Sources/TuistKit/GraphMappers/AutomationProjectMapperProvider.swift

```diff
@@ -37,6 +40,12 @@ final class AutomationProjectMapperProvider: ProjectMapperProviding {
             SourceRootPathProjectMapper()
         )
 
+        if skipUITests {
+            mappers.append(
+                PruneUITestsProjectMapper()
+            )
+        }
```

Comment:
Wonder if instead we should just always add the mapper and pass it the flag instead, the mapper would then decide what to prune and what not to prune?

Reply:
I'd keep the mappers simple, meaning they always perform the logic and the client decides when to use them.

Reply:
I'm inclined towards @mollyIV's suggestion here. The pattern that we've followed so far is letting the provider add/remove mappers depending on the flags and the project configuration. It's been working fine so I'd be consistent and not reconsider that as part of this PR.

Reply:
Sounds good, thanks for clarifying!